test/uk/gov/hmrc/hec/controllers/TaxCheckControllerWithInternalAuthEnabledSpec.scala

```diff
@@ -284,23 +328,43 @@ class TaxCheckControllerSpec extends ControllerSpec with AuthSupport {
       "return a 400 (bad request)" when {
 
         "the JSON in the request cannot be parsed" in {
-          status(performAActionWithJsonBody(JsString("hi"))) shouldBe BAD_REQUEST
-
+          mockInternalAuth(expectedPredicate)(Future.successful(Unit))
+          status(performAActionWithJsonBodyAndHeader(JsString("hi"))) shouldBe BAD_REQUEST
         }
 
       }
 
       "return an 500 (internal server error)" when {
 
         "there is an error saving the tax check" in {
+          mockInternalAuth(expectedPredicate)(Future.successful(Unit))
           mockMatchTaxCheck(individualMatchRequest)(Left(Error(new Exception("Oh no!"))))
 
-          val result = performAActionWithJsonBody(Json.toJson(individualMatchRequest))
+          val result = performAActionWithJsonBodyAndHeader(Json.toJson(individualMatchRequest))
           status(result) shouldBe INTERNAL_SERVER_ERROR
         }
 
       }
 
+      "return a UpstreamErrorResponse if the token is invalid" when {
+
+        "there is no authenticated session" in {
+          mockInternalAuth(expectedPredicate)(
+            Future.failed(UpstreamErrorResponse.apply("Unauthorized", Status.UNAUTHORIZED))
+          )
+          assertThrows[UpstreamErrorResponse](
```

Comment:
we could do a stonger assertion here and below by doing something like
```scala
val error = intercept[UpstreamErrorResponse](await(...))
error.statusCode shouldBe UNAUTHORIZED
```

Reply:
[updated](https://github.com/hmrc/hec/pull/55/commits/10e715e1fa59f26855ca13265e6a1a0987a50460)